ci/jenkinsfiles/common-lib.groovy

```diff
@@ -0,0 +1,402 @@
+        }
+      }
+    }
+  }
+}
+
+void runFunctionalTests(String ftestFolder, String namespace) {
+  echo '''
+    ----------------------------------------
+    Run "default" functional tests
+    ----------------------------------------
+  '''
+  sh """
+    cd ${ftestFolder}
+    npm run ftest -- --nuxeoUrl=http://preview.${namespace}.svc.cluster.local/nuxeo
+  """
+}
+
+void scaleResource(String namespace, String resourceType, String resourceName, String replicas) {
+  sh "kubectl -n ${namespace} scale ${resourceType} ${resourceName} --replicas=${replicas}"
+}
+
+void setup() {
+  sh '''
+    # create the Git credentials
+    jx step git credentials
+    git config credential.helper store
+  '''
+  env.MAVEN_ARGS = this.mavenArgs()
+}
+
+void setSlackBuildStatus(String channel, String message, String color) {
+  if ( env.DRY_RUN != 'true' ) {
+    slackSend(channel: channel, color: color, message: message)
+  }
+}
+
+void setGitHubBuildStatus(String context, String message, String state, String gitRepo) {
+  if ( env.DRY_RUN != 'true' && ENABLE_GITHUB_STATUS == 'true') {
+    step([
+      $class: 'GitHubCommitStatusSetter',
+      reposSource: [$class: 'ManuallyEnteredRepositorySource', url: gitRepo],
+      contextSource: [$class: 'ManuallyEnteredCommitContextSource', context: context],
+      statusResultSource: [
+        $class: 'ConditionalStatusResultSource', results: [[$class: 'AnyBuildResult', message: message, state: state]]
+      ],
+    ])
+  }
+}
+
+void setLabels() {
+  echo '''
+    ----------------------------------------
+    Set Kubernetes resource labels
+    ----------------------------------------
+  '''
+  echo "Set label 'branch: ${BRANCH_NAME}' on pod ${NODE_NAME}"
+  sh "kubectl label pods ${NODE_NAME} branch=${BRANCH_NAME}"
+  // output pod description
+  echo "Describe pod ${NODE_NAME}"
+  sh "kubectl describe pod ${NODE_NAME}"
+}
+
+void uploadPackage(String version, String credential,  String connectUrl) {
+  withCredentials(
+    [usernameColonPassword(credentialsId: "${credential}", variable: 'CONNECT_PASS')]
+  ) {
+    sh """
+      PACKAGE="nuxeo-coldstorage-package/target/nuxeo-coldstorage-package-${version}.zip"
+      curl -i -u "$CONNECT_PASS" -F package=@\$PACKAGE "$connectUrl"/site/marketplace/upload?batch=true
+    """
+  }
+}
+
+void updateVersion(String version) {
+  echo """
+    ----------------------------------------
+    Update version
+    ----------------------------------------
+    New version: ${version}
+  """
+  sh """
+    mvn ${MAVEN_ARGS} versions:set -DnewVersion=${version} -DgenerateBackupPoms=false   
+  """
+}
+
+/**
+ * Replaces environment variables present in the given yaml file and then runs skaffold build on it.
+ * Needed environment variables are generally:
+ * - DOCKER_REGISTRY
+ * - VERSION
+ */
+void skaffoldBuild(String skaffoldFile) {
+  sh """
+    envsubst < ${skaffoldFile} > ${skaffoldFile}~gen
+    skaffold build -f ${skaffoldFile}~gen
+  """
+}
+
+return this;
```

Comment:
sorry @PGoncalvesNX perhaps I am missing somethings, why we need this file. I mean we have already the unit test on front end part on the LTS-2021

Reply:
It was already removed before your comment 😅